Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -379,7 +379,7 @@ static inline void GenerateVertexShader(T& out, u32 components, API_TYPE api_typ
 
 	//write the true depth value, if the game uses depth textures pixel shaders will override with the correct values
 	//if not early z culling will improve speed
-	if (api_type == API_D3D)
+	if (api_type == API_D3D || g_ActiveConfig.backend_info.bSupportsClipControl)
```

Comment:
I'd just force-enable this option for D3D, so there won't be any need to check for the api_type here.